### ISSUE TITLE
Add `Triggers` table to metadata store

### ISSUE THREAD

**Issue by mdo6180** [anacostiaAI/anacostia-pipeline#66]
- [x] Add SQL statement to make Triggers table. Triggers table contains the following parameters:
  - [x] node_name (the name of the node that triggered the pipeline)
  - [x] run_triggered (the run it triggered, i.e., run_id + 1, i.e., the next run)
  - [x] timestamp (when trigger was triggered; sometimes the trigger happened during the current run but the current run takes a fairly long time to execute)
  - [x] message (the message describing why the pipeline was triggered, e.g., "loss = 3.5, trigger condition loss > 2 was satisfied"
- [x] change `.trigger()` method to `.trigger(message=None)`. do this for the `.trigger()` method in the following classes:
  - [x] BaseResourceNode
  - [x] BaseMetadataStoreNode
  - [x] SqliteMetadataStoreNode
- [x] add `.log_trigger(node_name: str, run_triggered: int, message: str = None)` method to `BaseMetadataStoreNode` and `SqliteMetadataStoreNode`
- [x] create GUI for Triggers table
- [x] figure out a way for resource nodes to call `.log_trigger()` on leaf pipeline
  - [x] define `.trigger()` as async 
  - [x] convert all functions that call `.trigger()` to async
  - [x] create RPC for `.log_trigger()`
